content/client-lib-development-guide/features.textile

```diff
@@ -107,6 +107,7 @@ h3(#rest-auth). Auth
 *** @(RSA4c1)@An @ErrorInfo@ with @code@ @80019@ and description of the underlying failure should be emitted with the state change, in the @errorReason@ and/or in the callback as appropriate
 *** @(RSA4c2)@If the connection is @CONNECTING@, then the connection attempt should be treated as unsuccessful, and as such the connection should transition to the @DISCONNECTED@ or @SUSPENDED@ state as defined in "RTN14":#RTN14 and "RTN15":#RTN15
 *** @(RSA4c3)@If the connection is @CONNECTED@, then the connection should remain @CONNECTED@
+** @(RSA4d)@ If an attempt to authenticate is made using the @authUrl@ and the content type is neither text nor JSON, or if using @authCallback@ and the object passed in is neither a @String@, @JsonObject@, @TokenRequest@ object nor a @TokenDetails@ object, the auth attempt should fail with error code @40143@ and a suitable error message.  Additionally, if the text token string is greater than 384 bytes, or the JSON stringified @JsonObject@, @TokenRequest@ or @TokenDetails@ is greater than 128kb, then the auth attempt should fail with error code @40143@ and a suitable error message.
```

Comment:
> Why did we did Token*.fromJson then?

Because it's a convenience and allows them to be explicit, but also because we can then simply call that method directly for customers?

I can add it to the IDL, will do now.